src/functional/swing_slide_filter.zig

```diff
@@ -278,10 +526,55 @@ fn appendValue(comptime T: type, value: T, compressed_values: *std.ArrayList(u8)
     }
 }
 
-/// Computes the intercept coefficient of a linear function that passes through the `point` with
-/// the given `slope` coefficient.
-fn computeInterceptCoefficient(slope: f80, point: DiscretePoint) f80 {
-    return point.value - slope * usizeToF80(point.time);
+/// Computes the intercept coefficient of a linear function that passes through the `DiscretePoint`
+/// `point` with the given `slope` coefficient.
+fn computeInterceptCoefficient(slope: f80, comptime point_type: type, point: point_type) f80 {
+    if (point_type == DiscretePoint) {
+        return point.value - slope * usizeToF80(point.time);
+    } else {
+        return point.value - slope * point.time;
+    }
+}
+
+/// Updates the linear function coeficients in `linear_function` that passes throught the two
+/// points of the `segment` slided up/down at the start point and down/up at the end point based
+/// on the `error_bound`. Specifically, the `linear_function` will pass through the points
+/// (`segment.start_point.time`, `segment.start_point.value - error_bound`) and
+/// (`segment.end_point.time`, `segment.end_point.value + error_bound`).
+fn updateSlideLinearFunction(
+    segment: Segment,
+    linear_function: *LinearFunction,
+    error_bound: f32,
+) void {
+    if (segment.end_point.time != segment.start_point.time) {
+        const duration: f80 = @floatFromInt(segment.end_point.time - segment.start_point.time);
+        linear_function.slope = (segment.end_point.value + 2 * error_bound -
+            segment.start_point.value) / duration;
+        linear_function.intercept = segment.start_point.value - error_bound -
+            linear_function.slope * usizeToF80(segment.start_point.time);
+    } else {
+        linear_function.slope = 0.0;
+        linear_function.intercept = segment.start_point.value;
+    }
+}
+
+/// Computes the interception point between `linear_function_1` and `linear_function_2` and
```

Comment:
The names of the parameters in the comment does not match the name of the parameters.